### ISSUE TITLE
order of extend clauses can lead to false positives of undefined method

### ISSUE THREAD

**Issue by robertclancy** [YorickPeterse/ruby-lint#156]
This example:
```ruby
module A
  def hello
    'hello'
  end
  extend self
end

module B
  extend self
  def hi
    'hi'
  end
end

puts A.hello
puts B.hi
```
with version ruby-lint v2.0.5 on ruby 2.0.0p645 outputs:
> test.rb: error: line 16, column 6: undefined method hi on B

Note: it complains about the `B` module but not `A`.

**Comment by YorickPeterse** [comment]
Correct, ruby-lint processes code in the order that it appears in. There's currently no way around this.
